### ISSUE TITLE
Ogg video file fails to import with ERR_FILE_CORRUPT

### ISSUE THREAD

**Issue by mnn** [godotengine/godot#56895]
### Godot version

v3.4.2.stable.official [45eaa2daf]

### System information

Linux (Manjaro), GLES3

### Issue description

Several video files (all captured by SimpleScreenRecorder) fail to import in Godot. Manual reimports lead to the same error. Those video files are working fine in SMPlayer, QtAV and VLC.

```
<> file simplescreenrecorder-2022-01-18_11.13.58.ogg
simplescreenrecorder-2022-01-18_11.13.58.ogg: Ogg data, Theora video
<> ffprobe simplescreenrecorder-2022-01-18_11.13.58.ogg
ffprobe version n4.4.1 Copyright (c) 2007-2021 the FFmpeg developers
  built with gcc 11.1.0 (GCC)
  configuration: --prefix=/usr --disable-debug --disable-static --disable-stripping --enable-amf --enable-avisynth --enable-cuda-llvm --enable-lto --enable-fontconfig --enable-gmp --enable-gnutls --enable-gpl --enable-ladspa --enable-libaom --enable-libass --enable-libbluray --enable-libdav1d --enable-libdrm --enable-libfreetype --enable-libfribidi --enable-libgsm --enable-libiec61883 --enable-libjack --enable-libmfx --enable-libmodplug --enable-libmp3lame --enable-libopencore_amrnb --enable-libopencore_amrwb --enable-libopenjpeg --enable-libopus --enable-libpulse --enable-librav1e --enable-librsvg --enable-libsoxr --enable-libspeex --enable-libsrt --enable-libssh --enable-libsvtav1 --enable-libtheora --enable-libv4l2 --enable-libvidstab --enable-libvmaf --enable-libvorbis --enable-libvpx --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxcb --enable-libxml2 --enable-libxvid --enable-libzimg --enable-nvdec --enable-nvenc --enable-shared --enable-version3
  libavutil      56. 70.100 / 56. 70.100
  libavcodec     58.134.100 / 58.134.100
  libavformat    58. 76.100 / 58. 76.100
  libavdevice    58. 13.100 / 58. 13.100
  libavfilter     7.110.100 /  7.110.100
  libswscale      5.  9.100 /  5.  9.100
  libswresample   3.  9.100 /  3.  9.100
  libpostproc    55.  9.100 / 55.  9.100
Input #0, ogg, from 'simplescreenrecorder-2022-01-18_11.13.58.ogg':
  Duration: 00:00:04.54, start: 0.000000, bitrate: 81 kb/s
  Stream #0:0: Video: theora, yuv420p, 550x500 [SAR 1:1 DAR 11:10], 24 tbr, 24 tbn, 24 tbc
    Metadata:
      encoder         : Lavf58.76.100
```

### Steps to reproduce

1. Create new project
2. Copy ogg file from [simplescreenrecorder-2022-01-18_11.13.58.zip](https://github.com/godotengine/godot/files/7888182/simplescreenrecorder-2022-01-18_11.13.58.zip) (decompress zip first) to project directory
3. See red cross in FileSystem dock and error in Output tab:
```
 modules/stb_vorbis/audio_stream_ogg_vorbis.cpp:189 - Condition "ogg_stream == nullptr" is true.
 modules/stb_vorbis/resource_importer_ogg_vorbis.cpp:94 - Condition "!ogg_stream->get_data().size()" is true. Returned: ERR_FILE_CORRUPT
 editor/editor_file_system.cpp:1755 - Error importing 'res://simplescreenrecorder-2022-01-18_11.13.58.ogg'.
```

**Comment by KoBeWi** [comment]
That's because Godot recognizes files only by extension. Change it to .ogv instead.

**Comment by akien-mga** [comment]
Indeed, and this is documented already:

https://docs.godotengine.org/en/stable/classes/class_videostreamtheora.html

> **Note:** While Ogg Theora videos can also have an `.ogg` extension, you will have to rename the extension to `.ogv` to use those videos within Godot.

**Comment by mnn** [comment]
And how a user is supposed to know that Godot doesn't support ogg, but does ogv? Only thing I see is the error message (without a link to docs) which just says it's corrupted which is untrue, the ogg and the video inside is perfectly fine and seems to be according to the standard. ogg supports audio and/or video (wiki):

> The Ogg container format can multiplex a number of independent streams for audio, video, text (such as subtitles), and metadata.

Godot should ideally support ogg, or at least say it doesn't support ogg with video, not just say the file is corrupted. Now it's a UX mess.

**Comment by akien-mga** [comment]
It says it's corrupted because `.ogg` is the most common extension for OGG Vorbis audio files, and since Godot only detects file types based on their extension, and `.ogg` file is handled as an OGG Vorbis file. Yours is not an OGG Vorbis file and thus can't be decoded by the OGG Vorbis importer.

Changing that would require very significant work to implement file type detection by actually parsing each file. That's a quite difficult to implement and here the workaround is clearly documented.

I can make the error message clearer about the parsed data not being a valid OGG Vorbis file though, that should maybe help understand the fact that it's being loaded by the wrong loader.

**Comment by akien-mga** [comment]
Fixed by #56903.
